openapi/clusters_mgmt/v1/openapi.json

```diff
@@ -14949,6 +14949,14 @@
           }
         }
       },
+      "AutoNodeMode": {
+        "description": "The configured mode of operation for AutoNode in the Cluster.",
+        "type": "string",
+        "enum": [
```

Comment:
I suggest to not keep it an enum as it limits us in expanding the "mode" if we need to (i.e. adding "enhanced" mode will break existing clients that do not know how to translate that enum).

Reply:
Documenting it in this repo, and that is auto generating the doc on openAPI spec

Reply:
@tzvatot Updated with API comments defining the allowed values. 

Note: I changed the field name to `state` to align with other resources on the API where the same `enabled`/`disabled` pattern is used.